src/tabmanager.cc

```diff
@@ -481,14 +481,18 @@ void TabManager::openTabFile(const QString &filename)
     const auto knownFileType = par->knownFileExtensions.contains(suffix);
     const auto cmd = par->knownFileExtensions[suffix];
     if (knownFileType && cmd.isEmpty()) {
-        setTabName(filename);
-        par->updateRecentFiles(editor);
+         setTabName(filename); //refeshDocument gets filename from editor
+         if (refreshDocument()) {     //doc opened
+            par->fileChangedOnDisk(); // force cached autoReloadId to update
+            par->updateRecentFiles(editor);
+        }
+        else {
+            setTabName(""); //can't open file
```

Comment:
I'm wondering if this case should still set the file name. This allows for `openscad new-file.scad` to give it a name already which will then be the default for save.

Reply:
I guess my thought was openscad new-file.scad doesn't actually open a file so it is misleading for the filename to be on the tab title bar. For example, openscad new-file.scad would act the same as openascad empty-file.scad even though the latter would open a real physically present file. 

Also, suppose you did this in a directory without write access. When you went to save it you would get an error (the next issue I was going to look at).

It is up to you. Let me know if you want me to make the change.

Thanks

Reply:
Hmm, good point, I'm not sure what the SaveFileDialog does when given a non-existent name. I'll need to have a look at that.